### PR TITLE
Wrapped stdErr and changed file open

### PR DESCRIPTION
**Pull request by bhourahine**
Also changed a few modules that should be using globalenv instead
of io definition of stdOut. Addresses most of #469.

### CODE REVIEW

File: prog/dftb+/lib_common/globalenv.F90 [hunk 1]
```diff
@@ -56,22 +62,31 @@ module dftbp_globalenv
...
 
     !> Customised global standard output
     integer, intent(in), optional :: outputUnit
 
     !> Customised global MPI communicator
     integer, intent(in), optional :: mpiComm
 
-    integer :: mpiComm0, outputUnit0
+    !> Customised global standard input
+    integer, intent(in), optional :: inputUnit
```
Comment (awvwgk) [inline prog/dftb+/lib_common/globalenv.F90:74]: This variable is not used, why have it in the dummy arguments?

**Comment by bhourahine** [comment]
> Can you remove the `dftb_io` module completely with this PR? Otherwise the changes are not really useful.

Unfortunately a few places need to know the stdOut unit before the comm world processor 0 is identified, so it can't be removed. But its use can be minimized.

**Comment by awvwgk** [comment]
> Unfortunately a few places need to know the stdOut unit before the comm world processor 0 is identified, so it can't be removed. But its use can be minimized.

Initializing `stdOut` with `stdOut0` and `stdErr` with `stdErr0` in `dftbp_globalenv` would solve this issue. Alternatively this places could directly use `stdOut0` and `stdErr0`.

**Comment by bhourahine** [comment]
> Initializing `stdOut` with `stdOut0` and `stdErr` with `stdErr0` in `dftbp_globalenv`

Good point.

File: prog/dftb+/lib_common/globalenv.F90 [hunk 2]
```diff
@@ -56,22 +62,28 @@ module dftbp_globalenv
...
 
-    integer :: mpiComm0, outputUnit0
+    !> Customised global standard error
+    integer, intent(in), optional :: errorUnit
+
+    integer :: mpiComm0, outputUnit0, errorUnit0
 
-    if (present(outputUnit)) then
-      outputUnit0 = outputUnit
+  #:for VAR, NAME in [("output", "stdOut"), ("error", "stdErr")]
```
Comment (awvwgk) [inline prog/dftb+/lib_common/globalenv.F90:78]: This preprocessor does not look to useful here, it just makes the code obscure.

**Pull request merged**